modules/apps/frontend-js/frontend-js-components-sample-web/src/main/resources/META-INF/resources/partials/feature_indicator.jsp

```diff
@@ -0,0 +1,99 @@
+<%--
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<clay:container-fluid>
+	<clay:row>
+		<clay:col>
+			<h2>React Component</h2>
+		</clay:col>
+	</clay:row>
+
+	<div>
+		<react:component
+			module="js/FeatureIndicatorSamples"
+			props='<%=
+				HashMapBuilder.<String, Object>put(
+					"learnResourceContext", LearnMessageUtil.getReactDataJSONObject("frontend-js-components-web")
+				).build()
+			%>'
+		/>
+	</div>
+
+	<clay:row>
+		<clay:col>
+			<h2>JSP Direct Instantiation of React Component</h2>
+		</clay:col>
+	</clay:row>
+
+	<clay:row>
+		<clay:col>
+			<h3>Beta Interactive</h3>
+
+			<react:component
+				module="{FeatureIndicator} from frontend-js-components-web"
```

Comment:
I don't think we should promote this way of referencing module. If I understood correctly, this is a legacy way of wiring, with hacky implementation under the hood, prone to problems with versioning. @izaera would know more details. Unrelated to PR, should we add a lint rule to prevent this?

 For the task of the PR, could we keep all samples within a React component wired like in the first sample,  `module="js/FeatureIndicatorSamples"`.

Reply:
@dsanz initially proposed this route with the thought that we should build a taglib for this to encapsulate it. How about we keep this for now and create a ticket to create the feature indicator taglib, at which point we can also migrate this and any other usages?

Reply:
What I fear is spreading of bad practice throughout DXP. There are currently only six usages like this in DXP, but if we don't stomp it out quickly, it'll be a costly tech debt migration down the line. People understandably tend to copy merged code that works.

I may be misunderstanding how harmful is this way of wiring. If it's harmless, but only not following concentions/ guidelines / best practices, then I'm ok with pushing it as a temporary solution. @dsanz @izaera what do you think?

Reply:
Almost all usages of `<react:component>` point to JS resources in the same module, which I'd say it's the expected pattern. When a react component has to be used from different modules, then situation is a bit awkward as we have to either use `module="{x} from other-place"` or create an alternative in the form of taglib as we do with clay tags.

Feature indicator will be used in many different context. I've identified 3 so far: from [react](https://github.com/liferay/liferay-portal/blob/397c09dfe885c1985b6dfcd8a77497202551c802/modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/components/render_times/RenderTimes.tsx#L69-L73), from [a JSP](https://github.com/liferay/liferay-portal/blob/97c385521ed949ace9aad733a2994aa2fcac0c2f/modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/admin/beta_client_extension_button.jsp#L11-L19) (useful to integrate with existing frameworks like [control menu entries](https://github.com/liferay/liferay-portal/blob/97c385521ed949ace9aad733a2994aa2fcac0c2f/modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/product/navigation/control/menu/BetaClientExtensionProductNavigationControlMenuEntry.java#L35)), or even [from Java](https://github.com/liferay/liferay-portal/blob/caee819bd1432f336369360d42d0780f1e5f7ad5/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java#L195-L202) in case one needs to emit markup from a fragment renderer. This one is essentially the same as using `<react:component>` but bypassing the taglib.

In terms of the module handling, I'd say react render infra already cares about modules like this and properly emits the corresponding imports to the page. The "gotcha" is that we need a esm-ready module to import things from, which is not a problem for `frontend-js-components-web` but it is for other modules. I'd say that's it though I'd like @izaera to confirm.

For the case of JSP I'd consider creating a taglib very much like the ones we already provide, probably in `liferay-frontend` namespace or perhaps a clay one (though this is not purely a clay component but a composition like the MT imho)

Reply:
> For the case of JSP I'd consider creating a taglib very much like the ones we already provide, probably in liferay-frontend namespace or perhaps a clay one (though this is not purely a clay component but a composition like the MT imho)

Yes, I agree. This shouldn't a Clay taglib since it's a composite.

Reply:
@markocikos That's the new de-facto standard syntax we use for ESM, not the old one you said we shouldn't use (that one does not contain `from` and looks like the AMD syntax but it doesn't specify the version number, which makes the infrastructure use some bridges we still have for very old legacy pre-AMD code :roll_eyes: ). 

React component tag is using `ReactRenderer` -> https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-taglib/frontend-taglib-react/src/main/java/com/liferay/frontend/taglib/react/servlet/taglib/ComponentTag.java#L47

which, in turn, checks for the syntax of the import here -> https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/portal-template/portal-template-react-renderer-impl/src/main/java/com/liferay/portal/template/react/renderer/internal/ReactRendererImpl.java#L46

Thus, that new syntax is the one we should really promote, subject to the caveats @dsanz explained above: basically that you can only use that syntax with OSGi bundles that have been migrated to ESM. That can be checked looking for a `main` or `exports` entry in the `npmscripts.config.js` file of the imported OSGi bundle, like in this one, for example -> https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-js/frontend-js-react-web/npmscripts.config.js

We should migrate everything to ESM in `liferay-portal` so that we don't need to care about 3 or 4 ways of loading modules (AUI, old AMD, new AMD, ESM...) but you know..., things go slow... :man_shrugging: